python/src/kestra.py

```diff
@@ -37,6 +44,229 @@ def timer(name, duration, tags=None):
+        """
+        self.wait_for_completion = wait_for_completion
+        self.poll_interval = poll_interval
+        self.labels_from_inputs = labels_from_inputs
+        self.user = os.environ.get("KESTRA_USER", None)
+        self.password = os.environ.get("KESTRA_PASSWORD", None)
+        self.hostname = os.environ.get("KESTRA_HOSTNAME", "http://localhost:8080")
+
+    def _make_request(self, method: str, url: str, **kwargs) -> requests.Response:
+        """
+        Make a request to the Kestra server, add 'auth' if username and password are set
+        :param method: can be 'get', 'post', 'put', 'delete', etc.
+        :param url: the URL of your Kestra server e.g. "http://localhost:8080"
+        :param kwargs: additional arguments to pass to the request e.g. json, files, etc.
+        :return: the response from the server
+        """
+        if self.user is not None and self.password is not None:
+            kwargs["auth"] = (self.user, self.password)
+        response = requests.request(method, url, **kwargs)
+        if response.status_code == 401:
+            raise Exception(
+                "Authentication required but not provided. Please set the username and password."
+            )
+
+        response.raise_for_status()
+        return response
+
+    def check_status(self, execution_id: str) -> requests.Response:
+        """
+        Check the status of the execution
+        :param execution_id: the ID of the execution
+        :return: the response from the server
+        """
+        return self._make_request(
+            "get",
+            self.hostname
+            + Flow.API_ENDPOINT_EXECUTION_STATUS.replace(
+                "PARAM_EXECUTION_ID", execution_id
+            ),
+        )
+
+    def get_logs(self, execution_id: str) -> requests.Response:
+        """
+        Get the execution logs
+        :param execution_id: the ID of the execution
+        :return: the response from the server
+        """
+        return self._make_request(
+            "get",
+            self.hostname
+            + Flow.API_ENDPOINT_EXECUTION_LOG.replace(
+                "PARAM_EXECUTION_ID", execution_id
+            ),
+        )
+
+    def execute(
+        self,
+        namespace: str,
+        flow: str,
+        inputs: dict = None,
+    ) -> namedtuple:
+        if inputs is None:
+            inputs = {}
+
+        logging.debug(
+            "Starting a flow %s in the namespace %s with parameters %s",
+            flow,
+            namespace,
+            str(inputs),
+        )
+        result = namedtuple("FlowExecution", ["status", "log", "error"])
+        url_default = self.hostname + Flow.API_ENDPOINT_EXECUTION_CREATE.replace(
+            "PARAM_FLOW_ID", namespace + "/" + flow
+        )
+        if self.labels_from_inputs and len(inputs) > 0:
+            labels = "?"
+            for key in inputs:
+                value = inputs[key]
+                labels = f"{labels}&labels={key}:{value}"
```

Comment:
Also remove labels= "?"
```suggestion
            labels = "?" + "&".join([f"labels={key}:{value}" for key, value in inputs.items()])
```

Reply:
After that you can merge it